### PR TITLE
Adding defaults to Reference pages

### PR DESCRIPTION
**Pull request by cwickham**
WIP
Just a place to discuss with @cscheid

### CODE REVIEW

File: tools/reference.ts [hunk 1]
```diff
@@ -92,6 +94,7 @@ const readGroupOptions = (context: string, name: string): Array<Option> => {
         formats: formatsFromOptionSchema(optionSchema),
         contexts: optionSchema.tags?.contexts,
         engine: optionSchema.tags?.engine,
+        default: optionSchema?.default
```
Comment (cwickham) [inline tools/reference.ts:97]: I couldn't remember if this is what you added, but it seemed to do the trick.

File: tools/reference.ts [hunk 2]
```diff
@@ -156,7 +159,7 @@ const optionsForFormat = (format: string) => {
         ...group,
         options: group.options
           .filter(option => option.formats?.includes(format))
-          .map(option => ({ name: option.name, description: option.description }))
+          .map(option => ({ name: option.name, description: option.description, default: option?.default }))
```
Comment (cwickham) [inline tools/reference.ts:162]: This is what I changed to propagate defaults to the formats.

File: tools/reference.ts [hunk 3]
```diff
@@ -225,7 +228,8 @@ function readProjectProperties(props: { [name: string]: Record<string, unknown>
     .filter(key => props[key].hidden !== true)
     .map(key => ({
       name: key,
-      description: descriptions?.[key] || asDescriptionString(findVal(props[key], "description"))
+      description: descriptions?.[key] || asDescriptionString(findVal(props[key], "description")),
+      default: findVal(props[key], "default")
```
Comment (cwickham) [inline tools/reference.ts:232]: This is what I added to pick up some of the defaults for project options. Seems to grab some, but I'm not convinced I'm getting them all...

File: tools/reference.ts (hunk 1 above)
Comment (cscheid) [reply tools/reference.ts:97]: Yup, that looks right!

File: tools/reference.ts (hunk 2 above)
Comment (cscheid) [reply tools/reference.ts:162]: Ah, that makes perfect sense, nice catch.

File: tools/reference.ts (hunk 3 above)
Comment (cscheid) [reply tools/reference.ts:232]: This looks good, though!

**Pull request closed without merging**